presentation/src/main/java/com/gtxtreme/template/presentation/home/HomeFragment.kt

```diff
@@ -20,11 +24,21 @@ class HomeFragment :
         viewModel.onIntent(HomeScreenIntent.Search)
     }
 
+    private val listComponent by component {
+        ListComponent(viewModel, recyclerViewId = R.id.homeRecyclerView) {
+            addRenderer(UIContentFavouriteListRenderer())
+        }
+    }
+
     override fun onState(screenState: HomeScreenState) {
         super.onState(screenState)
-
-        binding?.apply {
-            descriptionText.setUIText(screenState.text)
+        if (screenState.items.size() == 0) {
+            binding?.emptyText?.show()
+            binding?.homeRecyclerView?.hide()
+        } else {
+            binding?.emptyText?.hide()
+            binding?.homeRecyclerView?.show()
+            listComponent.setData(screenState.items)
```

Comment:
You can avoid using ? in all places by using binding?.apply { }

Also we can use 
val isListEmpty = screenState.items.size() == 0
emptyText.setVisible(isListEmpty)
recyclerView.setVisible(!isListEmpty)
```suggestion
       binding?.apply {
            val isListVisible = screenState.items.size() == 0
            emptyText.setVisible(isListVisible)
            homeRecyclerView.setVisible(!isListVisible)
            listComponent.setData(screenState.items)
        }
```